common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -923,4 +1063,13 @@ public static String getClosestSuggestionsOnInvalidName(
 
     return String.join(", ", suggestions);
   }
+
+  public static List<CollationIdentifier> listCollations(String catalog, String schema) {
+    return Collation.CollationSpec.listCollations(catalog, schema);
```

Comment:
I would maybe say lets keep catalog and schema outside of CollationFactory. How I see it is that we have sessionCatalog which has some catalog/schema defined and it is listing collations that it has under it. This way we keep Collations simple objects that only depend on the actual info they need to be constructed.

Reply:
Method `listCollations(String catalog, String schema)` can be changed to `listCollations()`,
But are we also changing the method loadCollation(String catalog, String schema)? Because `catalog` and `schema` will exist as an field of `CollationMeta`, if we don't pass it in here, we will need to build a new object outside, such as `CollationMetaAddon`, and then convert `CollationMeta` to `CollationMetaAddon`
`CollationMetaAddon` = `CollationMeta` + catalog + schema

Reply:
Maybe we can than leave this hardcoded in buildCollationMeta to be SYSTEM/BUILTIN until we see a need to actually use catalog and schema? I am not sure if for example in future catalog/schema information would be included in the name for user defined collations, so we should probably keep this simple for now, so we do not create additional work to migrate later.

Reply:
Okay, let's remove parameter passing and hardcoded it for now